Cargo.toml

```diff
@@ -24,6 +27,9 @@ control-buffer-256 = []
 # speed device, but the descriptors will be invalid.
 test-class-high-speed = []
 
+# Add defmt Format support for enums and structs
+defmt-impl = ["defmt"]
```

Comment:
Let's copy how `smoltcp` does this. I don't think we need to add an explicit feature flag. The user can just do `features=["defmt"]` and that should be sufficient.
https://github.com/smoltcp-rs/smoltcp/blob/cae682e0885502cc3fa4e122d44f67167addb095/Cargo.toml#L24

https://github.com/smoltcp-rs/smoltcp/blob/cae682e0885502cc3fa4e122d44f67167addb095/src/socket/icmp.rs#L25